### PR TITLE
New recipe: Soup3

### PR DESCRIPTION
**Pull request by lukebemish**
Here's my attempt to get libsoup3 bundled, assuming I didn't mess anything up locally. I based this off some of the scripts in https://github.com/JuliaPackaging/Yggdrasil/pull/5689.

### CODE REVIEW

File: S/Soup3/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    -Ddocs=disabled \
+    -Dsysprof=disabled \
+    ..
+
+if [[ "${target}" != *darwin* ]]; then
+    # https://github.com/rbgirshick/py-faster-rcnn/issues/706
+    sed -i "s/-R/-Wl,-rpath=/" build.ninja
+fi
+
+if [[ "${target}" != *darwin* && "${target}" != *w64* ]]; then
```
Comment (giordano) [inline S/Soup3/build_tarballs.jl:31]: I don't think this is very accurate:
```suggestion
if [[ "${target}" == *86*-linux-gnu* ]]; then
```

File: S/Soup3/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    # https://stackoverflow.com/questions/2418157/c-error-undefined-reference-to-clock-gettime-and-clock-settime
+    sed -i "s/\$ARGS -o \$out \$in \$LINK_ARGS/\$ARGS -o \$out \$in \$LINK_ARGS -lrt/" build.ninja
```
Comment (giordano) [inline S/Soup3/build_tarballs.jl:33]: If you use single quotes you don't have to escape every single dollar sign, and this becomes much more readable:
```suggestion
    sed -i 's/$ARGS -o $out $in $LINK_ARGS/$ARGS -o $out $in $LINK_ARGS -lrt/' build.ninja
```

File: S/Soup3/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+products = [
+    LibraryProduct(["libsoup", "libsoup-3", "libsoup-3.0"], :libsoup),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    # Host gettext needed for "msgfmt"
+    HostBuildDependency("Gettext_jll"),
+    # Gettext is only needed on macOS
+    Dependency("Gettext_jll", v"0.21.0"; compat="=0.21.0", platforms=filter(Sys.isapple, platforms)),
```
Comment (giordano) [inline S/Soup3/build_tarballs.jl:54]:
```suggestion
    Dependency("Gettext_jll"; compat="=0.21.0", platforms=filter(Sys.isapple, platforms)),
```
Also, are you sure we cannot avoid it at all?  Gettext is a terrible dependency to have

File: S/Soup3/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    Dependency("Glib_jll"; compat="2.74.0"),
+    Dependency("SQLite_jll"),
+    Dependency("nghttp2_jll"),
+    Dependency("brotli_jll"),
+    Dependency("libpsl_jll"),
+    Dependency("LibUnwind_jll"),
```
Comment (giordano) [inline S/Soup3/build_tarballs.jl:60]: Are you sure this is really needed?  Libunwind isn't a common dependency.

File: S/Soup3/build_tarballs.jl (hunk 2 above)
Comment (lukebemish) [reply S/Soup3/build_tarballs.jl:33]: I'm pretty sure variable interpolation doesn't work inside single quotes, though it's been a while

Comment (giordano) [reply S/Soup3/build_tarballs.jl:33]: That's the point.

Comment (lukebemish) [reply S/Soup3/build_tarballs.jl:33]: Oh wait, I misread that line. Thanks

**Pull request merged**